### PR TITLE
elf,ar: Add ar decoder, improved elf decoder

### PR DESCRIPTION
**Pull request by wader**

### CODE REVIEW

**Comment by wader** [comment]
Maybe diasm should be split out

**Comment by Doctor-love** [comment]
Nice to see AR support - was thinking of working on it myself!

File: format/ar/ar.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
+package isa
```
Comment (wader) [inline format/ar/ar.go:1]: Wrong package name

**Comment by wader** [comment]
> Nice to see AR support - was thinking of working on it myself!

Aha, i noticed that ELFs seems to be hinding in them!

**Pull request merged**